app/src/main/java/com/grzegorzdec/tagmango/common/recyclerview/ListBinder.kt

```diff
@@ -0,0 +1,36 @@
+package com.grzegorzdec.tagmango.common.recyclerview
+
+import android.os.Handler
+import android.os.Looper
+import androidx.recyclerview.widget.DiffUtil
+
+
+class ListBinder<E>(private val diffCallBack: AbstractDiffCallback<E>) {
+    private var current: List<E> = ArrayList()
+    private var onDataChangeListener: OnDataChangeListener? = null
+
+    internal interface OnDataChangeListener {
+        fun onChange(diffResult: DiffUtil.DiffResult)
+    }
+
+    internal fun setOnDataChangeListener(onDataChangeListener: OnDataChangeListener) {
+        this.onDataChangeListener = onDataChangeListener
+    }
+
+    fun notifyDataChange(data: List<E>?) {
+        Handler(Looper.getMainLooper()).post {
+            val diffResult = calculateDiff(data)
+            if (onDataChangeListener != null) {
+                onDataChangeListener!!.onChange(diffResult)
```

Comment:
why `!!` instead of `?` I know in this case it does not matter, but it is a good approach to avoid `!!` fully.

Reply:
You're right, I think I didn't notice the autocompletion